cloudinit/stages.py

```diff
@@ -403,6 +404,24 @@ def _reflect_cur_instance(self):
 
         # Write out information on what is being used for the current instance
         # and what may have been used for a previous instance...
+        if getattr(self, "_netcfg"):
```

Comment:
Nit: since we've already initialized it, `if self._netcfg:` should be fine and a little clearer.
Nit2: This function is getting a bit long. Since this behavior is already self-contained, does it make sense as a separate function call?

Reply:
You are correct here, But, I'm dropping this as approach and going back to my original because `apply_network_config` is called in 3 circumstances:
 1. In Local stage after Local datasource is detected (instancify already called at ds detection)
 2. In Local stage before Network datasource is detected to render basic net config on primary NIC (instancify not yet called)
 3. In Network stage after a Network datasource is detected (instancify already called at ds detection)

In all cases, apply_network_config will not have been called prior to instancify, so no preserved _net_cfg is yet set.